### PR TITLE
Skip hidden dropdowns while focusing

### PR DESCRIPTION
**Pull request by jeremyvii**
This PR fixes issue #29405 by filtering out dropdown items set to `display: none` when focusing between items. The `isVisible()` utility method filters out the hidden items.

### CODE REVIEW

**Comment by patrickhlauke** [comment]
not had a chance to test, but i'll just throw out that you might also need to check for visible, but disabled, ones as well

**Comment by jeremyvii** [comment]
I think disabled items are ignored per this dropdown selector [here](https://github.com/twbs/bootstrap/blob/master/js/src/dropdown.js#L67), but I will double check.

**Review by Johann-S — CHANGES_REQUESTED** [review]
Hi @jeremyvii,

Thanks for your PR!
Can you add a unit test please ?

**Comment by jeremyvii** [comment]
While adding this unit test, I had noticed that `isVisible()` doesn't account for the `d-none` class. I added a check for this class in `isVisible()`.

**Comment by jeremyvii** [comment]
I was just reviewing the change I made, and think it may be better to use `getComputedStyle()` for checking if an element is hidden. Rather than checking for the `d-none` class explicitly. That way if a user has some rouge class that hides an element it will still be skipped while focusing between items.

File: js/tests/unit/dropdown.spec.js [hunk 1]
```diff
@@ -1292,6 +1292,43 @@ describe('Dropdown', () => {
...
+
+      const triggerDropdown = fixtureEl.querySelector('[data-toggle="dropdown"]')
+      const dropdown = fixtureEl.querySelector('.dropdown')
+
+      dropdown.addEventListener('shown.bs.dropdown', () => {
+        const keyDown = createEvent('keydown')
+        keyDown.which = 40
+
+        triggerDropdown.dispatchEvent(keyDown)
+        triggerDropdown.dispatchEvent(keyDown)
```
Comment (Johann-S) [inline js/tests/unit/dropdown.spec.js]: why did you dispatched the same events twice ?

Comment (jeremyvii) [reply js/tests/unit/dropdown.spec.js]: That's an oversight on my part. I have removed the extra `dispatchEvent()` in the latest commit.

**Review by Johann-S — APPROVED** [review]
